benchmarks/benchmarks/ci/esmf_regridder.py

```diff
@@ -151,3 +158,213 @@ def time_lazy_regridding(self, cache):
+
+    def setup(self, type):
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        n_lons_src = 20
+        n_lats_src = 40
+        n_lons_tgt = 20
+        n_lats_tgt = 40
+        h = 100
+        if type == "large source":
+            n_lons_src = 100
+            n_lats_src = 200
+        if type == "large target":
+            n_lons_tgt = 100
+            n_lats_tgt = 200
+        if type == "mixed":
+            coord_system_src = RotatedGeogCS(0, 90, 90)
+        else:
+            coord_system_src = None
+        mesh = _gridlike_mesh(n_lons_src, n_lats_src)
+        tgt = _grid_cube(
+            n_lons_tgt,
+            n_lats_tgt,
+            lon_bounds,
+            lat_bounds,
+            coord_system=coord_system_src,
+        )
+        src_data = np.arange(n_lats_src * n_lons_src * h).reshape([-1, h])
+        src = Cube(src_data)
+        mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+        src.add_aux_coord(mesh_coord_x, 0)
+        src.add_aux_coord(mesh_coord_y, 0)
+        self.regridder = MeshToGridESMFRegridder(src, tgt)
+        self.src = src
+
+    def time_perform_regridding(self, type):
+        _ = self.regridder(self.src)
+
+
+class TimeLazyMeshToGridRegridding:
+    # Prevent repeat runs between setup() runs - data won't be lazy after 1st.
+    number = 1
+    # Compensate for reduced certainty by increasing number of repeats.
+    #  (setup() is run between each repeat).
+    #  Minimum 5 repeats, run up to 30 repeats / 20 secs whichever comes first.
+    repeat = (5, 30, 20.0)
+    # Prevent ASV running its warmup, which ignores `number` and would
+    # therefore get a false idea of typical run time since the data would stop
+    # being lazy.
+    warmup_time = 0.0
+
+    def setup_cache(self):
+        SYNTH_DATA_DIR = Path().cwd() / "tmp_data"
+        SYNTH_DATA_DIR.mkdir(exist_ok=True)
+        file = str(SYNTH_DATA_DIR.joinpath("chunked_cube.nc"))
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        n_lons_src = 100
+        n_lats_src = 200
+        n_lons_tgt = 20
+        n_lats_tgt = 40
+        h = 2000
+        mesh = _gridlike_mesh(n_lons_src, n_lats_src)
+        tgt = _grid_cube(n_lons_tgt, n_lats_tgt, lon_bounds, lat_bounds)
+
+        chunk_size = [n_lats_src * n_lons_src, 10]
+        src_data = da.ones([n_lats_src * n_lons_src, h], chunks=chunk_size)
+        src = Cube(src_data)
+        iris.save(src, file, chunksizes=chunk_size)
+        # Construct regridder with a loaded version of the grid for consistency.
+        loaded_src = iris.load_cube(file)
+        # While iris is not able to save meshes, we add these after loading.
+        mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+        loaded_src.add_aux_coord(mesh_coord_x, 0)
+        loaded_src.add_aux_coord(mesh_coord_y, 0)
+        regridder = MeshToGridESMFRegridder(loaded_src, tgt)
+
+        return regridder, file, mesh
+
+    def setup(self, cache):
```

Comment:
Put in a `# TODO:` to come back once Iris can save a `Mesh`, getting rid of the confusing repetition.